extension-androidauto/src/main/java/com/mapbox/maps/extension/androidauto/MapboxCarMapSessionInstaller.kt

```diff
@@ -93,7 +96,21 @@ class MapboxCarMapSessionInstaller(
     return mapboxCarMap
   }
 
+  private fun enforceSharedContext(mapInitOptions: MapInitOptions) {
+    if (mapInitOptions.mapOptions.contextMode != ContextMode.SHARED) {
+      logW(
+        TAG,
+        "Explicitly switching to MapOptions.contextMode = ContextMode.SHARED when creating the car map surface. " +
+          "ContextMode.UNIQUE (used by default) is not allowed as it leads to graphical artifacts and crashes."
+      )
+      mapInitOptions.mapOptions = mapInitOptions.mapOptions.toBuilder()
+        .contextMode(ContextMode.SHARED)
```

Comment:
I tried doing this downstream in a navigation test app and it was not making a difference. https://github.com/mapbox/mapbox-navigation-android/blob/f528c3011438ba31c3383382697129fd9f581f74/android-auto-app/src/main/java/com/mapbox/navigation/examples/androidauto/car/MainCarSession.kt#L51

Are you sure this will fix it? Or are we looking into multiple issues?

Reply:
Oh I'm reading more conversations and it seems the behaviors are still different on Xiaomi devices. I think I was reproducing on Xiaomi, let me verify again.

Reply:
@kmadsen actually it turned out the issue reproduced on any device with Adreno inside. But with this PR it should work everywhere, `ContextMode.SHARED` make gl-native to clear everything after rendering, does it really not work for you with this fix?

Reply:
I've been testing for awhile on Xiaomi and it does make a difference. Wonder if my build was messed up yesterday when I tested. So this looks good!